### PR TITLE
Add more type annotations

### PR DESCRIPTION
**Pull request by zariiii9003**

### CODE REVIEW

File: cantools/typechecking.py [hunk 1]
```diff
@@ -1,27 +1,60 @@
...
-        "signals": typing.List["Signal"],
+        "signals": List["Signal"],
         "formats": Formats,
-        "multiplexers": typing.Dict[str, typing.Dict[int, typing.Any]],
+        "multiplexers": Dict[str, Dict[int, Any]],  # "Any" should be "Codec" (cyclic definition is not possible though)
     },
 )
+
+
+# Type aliases. Introduced to reduce type annotation complexity while
```
Comment (andlaus) [inline cantools/typechecking.py:42]: nice. I missed that file (`typechecking.py`) so  I put the type aliases into `message.py`. Thanks for cleaning up after me :)

**Review by andlaus — APPROVED** [review]
this is a meaningful improvement of the type annotation infrastructure and there no objections regarding the implementation from my end. @juleq?

**Comment by juleq** [comment]
Yes, @zariiii9003, thanks. Please tell me if it is ready to merge on your end.

**Comment by zariiii9003** [comment]
@juleq Yes it is ready, thanks

**Pull request merged**